internal/controller/controllers/infraenv_controller.go

```diff
@@ -341,6 +343,19 @@ func (r *InfraEnvReconciler) ensureISO(ctx context.Context, log logrus.FieldLogg
 	return r.updateInfraEnvStatus(ctx, log, infraEnv, updatedInfraEnv)
 }
 
+func ShouldEnableIronicAgent(log logrus.FieldLogger, infraEnv *aiv1beta1.InfraEnv) bool {
```

Comment:
if it using ironic agent depends on service configuration and we verify it in ignition process , why it even matter which label we have for infraenv? I assume that we should allow both option and decide by looking on infraenv label and not service configuration

Reply:
As explained above, it's both.
There is a service wide configuration for enabling the converged flow, and there is an infraEnv configuration whether to enable the Ironic Agent (use the converge flow) in that InfraEnv ISO